index.js

```diff
@@ -20,9 +21,24 @@ function codeImport(options = {}) {
         continue;
       }
 
-      const filePath = fileMeta.slice('file='.length);
+      const res = /^file=(?<path>.+?)(?:#(?:L(?<from>\d+))?(?<dash>-)?)?(?:L(?<to>\d+))?$/.exec(fileMeta)
+      if (!res || !res.groups || !res.groups.path) {
+        throw new Error(`Unable to parse file path ${fileMeta}`)
+      }
+      const filePath = res.groups.path
+      const hasDash = !! res.groups.dash
+      const fromLine = res.groups.from ? parseInt(res.groups.from) : undefined
+      const toLine = res.groups.to ? parseInt(res.groups.to) : undefined
       const fileAbsPath = path.resolve(file.dirname, filePath);
 
+      const extractLines = (content, fromLine, toLine) => {
```

Comment:
Prefer to move this function outside the `codeImport` function and make it pure (side-effect-free) so that it's easier to reason about.

Reply:
Not quite sure what you mean with pure or what side-effect it was creating; it was always pure, even it defined inline (except for `hasDash` which it wasn't modifying). The args shadow the local vars so it's not even a closure. I extracted `extractLines` to the top level though if that's what you meant.

Reply:
Yeah, I just prefer it to be outside the main function so that it's a little bit more clear for me to review 😅.